pkg/provision/engine.go

```diff
@@ -36,7 +38,7 @@ type defaultEngine struct {
 	nodeID string
 	source ReservationSource
 	store  ReservationCache
-	fb     Feedbacker
+	cl     *client.Client
```

Comment:
Isn't this a bit sad that we loose the interface here ?

Reply:
It is, but we also need 2 different interfaces here. so It's either we pass 2 instances of interfaces (instead of one) or just pass one client

Reply:
Or we just define a Interface that embed the 2 interface we need ?

Reply:
I don't see the added benefit of that, that would be an interface with so many methods that we don't use. 
I mean if the provision engine requires a client to the explorer, i don't see this as an odd code. Plus the client itself, does not provide any functionality, it's just a carrier for the actual interfaces. I would just leave it like that